lib/rubocop/cop/style/trivial_accessors.rb

```diff
@@ -117,13 +117,19 @@ def accessor(kind, method_name)
         end
 
         def autocorrect(node)
-          if node.type == :def
+          if node.type == :def && !in_instance_context?(node)
             autocorrect_instance(node)
           elsif node.type == :defs && node.children.first.type == :self
             autocorrect_class(node)
           end
         end
 
+        def in_instance_context?(node)
+          (node.parent &&
+           node.parent.type == :block &&
+           node.parent.children.first.children.last == :instance_eval)
```

Comment:
I'm not fond of using array access to node elements - we favour the use of splatting.